### DISCUSSION TITLE
Multiple Servers

### DISCUSSION THREAD

**Discussion by Krischan101** [miguelgrinberg/microdot#189] [Q&A]
Hi everyone,
I was wondering if running multiple servers simultaneously is possible?

**Comment by miguelgrinberg** [comment]
Yes. If you use an ASGI or WSGI server you can choose as many workers as you need.

**Reply by Krischan101** [reply]
I see, but can I assign different port for each server? I checked the examples, but I couldn't find a way to specify the port.

**Reply by miguelgrinberg** [reply]
A different port for each server? If that's what you want then you can run individual instances, each assigned to the port of your choice. The common thing to do when running multiple workers is to attach all of them to the same port. If you use different ports then you need a load balancer in front to spread the traffic among your workers.

**Reply by Krischan101** [reply]
for my specific purpose, a load balancer is unnecessary because each server instance is serving different service.

this is how I able to run multiple servers on different port

```
app1 = Microdot()
app2 = Microdot()
app3 = Microdot()

@app1.route('/')
def home(request):
    return "<h2>Server 1</h2>", 200, {'Content-Type': 'text/html'}

@app2.route('/')
def home(request):
    return "<h2>Server 2</h2>", 200, {'Content-Type': 'text/html'}

@app3.route('/')
def home(request):
    return "<h2>Server 3</h2>", 200, {'Content-Type': 'text/html'}

async def serve():
    server1 = uasyncio.create_task(app1.start_server(port=80))
    server2 = uasyncio.create_task(app2.start_server(port=81))
    server3 = uasyncio.create_task(app3.start_server(port=82))
    await uasyncio.gather(server1, server2, server3)

uasyncio.run(serve())
```

**Reply by miguelgrinberg** [reply]
Yes, okay, I did not understand that you wanted to run different servers on different ports. What you are doing is actually correct.
